src/main/java/seedu/address/logic/commands/ScheduleCommand.java

```diff
@@ -0,0 +1,58 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.util.function.Predicate;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.model.Model;
+import seedu.address.model.student.Student;
+
+/**
+ * Finds and list all students with classes on a given date.
+ * This is also the schedule of the user on that particular date.
+ */
+public class ScheduleCommand extends Command {
+
+    public static final String COMMAND_WORD = "schedule";
+
```

Comment:
If we are finding class time with respect to dayofweek, then would that mean that the user input of a specific date is redundant? If we just convert the specific user input date in dayofweek, we could just ask user to input 1,2,3,4,5,6,7 (Integer day of week).

Reply:
Hmmm what I thought was that input a date is more flexible and convenient. For instance user might want to look up a his/her schedule on a particular date, our app would translate that date internally to the day of week so that the user wouldnt need to check up the calendar. I was thinking of an extension along this lines too, where user could input a string indicating the day or week. What do you think?

Reply:
Oh that is true also. It is easier for a user to remembers the current date rather than the specific day! If we assume this, then I think the current implementation is fine.